### PR TITLE
fix: overrides should not appear in the manifest cache key

### PR DESCRIPTION
**Pull request by crenshaw-dev**
Currently, [git-based parameter overrides](https://argo-cd.readthedocs.io/en/stable/user-guide/parameters/#store-overrides-in-git) appear in the manifest cache key.

That's a problem, because parameter overrides aren't known when the cache `get` occurs. Overrides _shouldn't_ be in the cache key, because getting parameter overrides is a repo operation, and the entire point of the cache is to avoid repo operations.

So this change makes sure that overrides do not become part of the cache key. Changes to overrides will still bust the cache, because the changes will cause a change in the `revision` component of the key.

**Case for cherry-picking:** we have a 2.4 user who's spending a lot of money on `git fetches`. I think this is a fairly safe change, and it would help them out.

### CODE REVIEW

File: reposerver/repository/repository.go [hunk 1]
```diff
@@ -985,7 +989,7 @@ func mergeSourceParameters(source *v1alpha1.ApplicationSource, path, appName str
 		overrides = append(overrides, filepath.Join(path, fmt.Sprintf(appSourceFile, appName)))
 	}
 
-	var merged v1alpha1.ApplicationSource = *source.DeepCopy()
+	var merged = *source.DeepCopy()
```
Comment (crenshaw-dev) [inline reposerver/repository/repository.go:992]: The explicit type was redundant.

**Review by leoluz — APPROVED** [review]
LGTM

**Comment by crenshaw-dev** [comment]
Cherry-picked onto release-2.4 for release with 2.4.3.

**Comment by arloliu** [comment]
I think the plugin environment modification should invalidate the cache too. (e.g. `argocd app set <app_name> --plugin-env ...`)
When we change the plugin environments on the fly, the current manifest doesn't be invalidated, and it will cause the app to stay in the synced state, the following app sync operation will not take the new environment variable.

**Comment by crenshaw-dev** [comment]
@arloliu I'm surprised that plugin env vars aren't breaking the cache. They're part of the `ApplicationSource` object. Is there an issue already created for the problem you're hitting?

**Comment by arloliu** [comment]
It's the problem I met these days.
I tested it with a simple shell script cmp and directory app.
the behavior is:
1. `argocd app set <app_name> --plugin-env TEST_ENV=test`
2. `argocd app sync <app_name>` 

the CMP will receive the `TEST_ENV` when executing `generate` procedure in step 2, but the app will not sync with the new manifest even CMP produces a new manifest.(e.g. change replicas number)

And if we use the following steps.
1. `argocd app set <app_name> --plugin-env TEST_ENV=test`
2. `argocd app get <app_name> --hard-refresh`
3. `argocd app sync <app_name>` 

 the cache will be invalidated in step 2, and step 3 will use the manifest that CMP generated, but the problem is, step 2 also cleans the plugin environment variable that step 1 set. 

The step 2 invokes CMP `generate`  3 times.

**Comment by crenshaw-dev** [comment]
Can you create an issue with those reprod. steps?

> BTW, the step 2 invokes CMP generate 3 times if the app's git repo had updated.

Can you create an additional issue for this? :-)

**Comment by arloliu** [comment]
sure, I'll create a issue for it.
the cache is a bit weird, it remembers all histories of plugin env.
that's said:
if we never set `TEST_ENV` plugin env.
1. set `TEST_ENV=a` it invokes CMP generate
2. set `TEST_ENV=b` it invokes CMP generate
3. set `TEST_ENV=a` again, it will not invoke CMP generate
But it should invoke CMP because `TEST_ENV value already changed.

**Comment by arloliu** [comment]
Thanks for your help, I fill the info. and created a issue here:
https://github.com/argoproj/argo-cd/issues/9844

**Pull request merged**